### ISSUE TITLE
dereferencing Number.MAX_VALUE crashes Frida

### ISSUE THREAD

**Issue by BLuFeNiX** [frida/frida#1851]
I have observed this issue on an x86_64 Android emulator, using the latest x86_64 version of Frida (15.1.1), after updating `frida-tools` with pip).

The simplest way to reproduce this issue is by typing `Number.MAX_VALUE` into the Frida console, like so:
```
$ frida -U -f com.example.app --no-pause
     ____
    / _  |   Frida 15.0.0 - A world-class dynamic instrumentation toolkit
   | (_| |
    > _  |   Commands:
   /_/ |_|       help      -> Displays the help system
   . . . .       object?   -> Display information about 'object'
   . . . .       exit/quit -> Exit
   . . . .
   . . . .   More info at https://frida.re/docs/home/
Spawned `com.example.app`. Resuming main thread!                          
[Android Emulator 5554::com.example.app]-> Number.MAX_VALUE
Process crashed: Bad access due to invalid address

***
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'google/sdk_gphone_x86_64_arm64/generic_x86_64_arm64:11/RSR1.201211.001.A1/7054069:userdebug/dev-keys'
Revision: '0'
ABI: 'x86_64'
Timestamp: 2021-09-08 11:58:33-0400
pid: 7695, tid: 7721, name: com.example.app  >>> com.example.app <<<
uid: 10153
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
Cause: null pointer dereference
    rax 00007f4ff0448800  rbx 0000000000000000  rcx 00007f4ff02df990  rdx 0000000000000020
    r8  00007f4ff0448800  r9  00007f4ff028f340  r10 00007f4ff028f2f0  r11 000000000002e100
    r12 0000000000000020  r13 0000000000000010  r14 00007f4ff0448800  r15 00007f4ff02df990
    rdi 0000000000000000  rsi 0000000000000010
    rbp 00007f4ff028f0e0  rsp 00007f4fa720ca68  rip 00007f4fa544599b
backtrace:
      #00 pc 0000000000ee699b  /data/local/tmp/re.frida.server/frida-agent-64.so
      #01 pc 0000000000000003  <unknown>
***

Thank you for using Frida!
```

The same behavior can be observed for things like `Java.use('java.lang.Double').MAX_VALUE.value`.

**Comment by BLuFeNiX** [comment]
This appears to be resolved in Frida 15.1.12 (perhaps even earlier, though I am unsure).

```
$ frida -U -n system_server
     ____
    / _  |   Frida 15.1.12 - A world-class dynamic instrumentation toolkit
   | (_| |
    > _  |   Commands:
   /_/ |_|       help      -> Displays the help system
   . . . .       object?   -> Display information about 'object'
   . . . .       exit/quit -> Exit
   . . . .
   . . . .   More info at https://frida.re/docs/home/
                                                                                
[Android Emulator 5554::system_server]-> Number.MAX_VALUE
1.7976931348623157e+308
[Android Emulator 5554::system_server]-> Java.use('java.lang.Double').MAX_VALUE.value
1.7976931348623157e+308
```

**Comment by oleavr** [comment]
Yeah this was fixed in 15.1.5. Thanks for checking back on this, and sorry for the lack of triage — spread a bit too thin these days.

**Comment by BLuFeNiX** [comment]
@oleavr Understandable! Thanks for the reply; I'm glad to see that you're paying attention to the issues.
